go.mod

```diff
@@ -3,6 +3,7 @@ module github.com/notaryproject/notation-go
 go 1.22.0
 
 require (
+	github.com/AdaLogics/go-fuzz-headers v0.0.0-20240806141605-e8a1dd7889d6
```

Comment:
This PR is bringing in a new 3rd party dependency to this library. 
@shizhMSFT @priteshbandi would like to know your thoughts regarding this change.

Reply:
New dependency should not be added. If `github.com/AdaLogics/go-fuzz-headers` is really required for testing, we should create a nested model like https://github.com/notaryproject/notation/tree/main/test/e2e

Reply:
@shizhMSFT Could you elaborate what a nested model is? AFAICT, `test/e2e` is a separate module with its own `go.mod`. Would that be a nested model?

Reply:
Yes, creating a separate module with its own `go.mod`.